odl/discr/lp_discr.py

```diff
@@ -188,6 +188,13 @@ def size(self):
         """Total number of underlying partition cells."""
         return self.partition.size
 
+    def __len__(self):
+        """Return ``len(self)``.
+
+        Total number of underlying partition cells.
+        """
+        return self.size
```

Comment:
This should return `len(self.partition)`, no? `DiscreteLp` "looks like" the partition it's defined on.